### PR TITLE
Lesson11

### PR DESCRIPTION
**Pull request by Naumych**
add RouteController and stuff

### CODE REVIEW

File: app/views/routes/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<h1>Edit Route</h1>
```
Comment (svidlonix) [inline app/views/routes/edit.html.erb]: Are you really create it manually without scaffold ? Or it's copy pasta ?

File: app/controllers/routes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+class RoutesController < ApplicationController
+  before_action :set_route, only: [:show, :edit, :update, :destroy]
+  def index
+    @routes = Route.all
+  end
+
+  def show
```
Comment (svidlonix) [inline app/controllers/routes_controller.rb]: please write in one line empty method

File: app/controllers/routes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @route = Route.new(route_params)
+
+    if @route.save
+      redirect_to @route
+    else
+      render :new
+    end
+  end
+
+  def edit
```
Comment (svidlonix) [inline app/controllers/routes_controller.rb]: please write in one line empty method.

File: app/controllers/routes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def update
+
```
Comment (svidlonix) [inline app/controllers/routes_controller.rb]: don't leave white space line

**Review by svidlonix — COMMENTED** [review]
Do you test it manually ?

File: app/views/routes/edit.html.erb (hunk 1 above)
Comment (Naumych) [reply app/views/routes/edit.html.erb]: It's copypasta just as in lesson

File: app/controllers/routes_controller.rb (hunk 4 above)
Comment (Naumych) [reply app/controllers/routes_controller.rb]: Fixed

File: app/controllers/routes_controller.rb (hunk 3 above)
Comment (Naumych) [reply app/controllers/routes_controller.rb]: Fixed

File: app/controllers/routes_controller.rb (hunk 2 above)
Comment (Naumych) [reply app/controllers/routes_controller.rb]: Fixed

File: app/models/route.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class Route < ActiveRecord::Base
+  validates :name, presence: true
+end
```
[the remark this answers is not in the document]
Comment (Naumych) [reply app/models/route.rb]: Fixed

File: app/controllers/routes_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  private
+
+  def set_route
+    @route = Route.find(params[:id])
+  end
+
+  def route_params
+    params.require(:route).permit(:name)
+  end
+end
```
[the remark this answers is not in the document]
Comment (Naumych) [reply app/controllers/routes_controller.rb]: Fixed

File: app/views/welcome/index.html.erb [hunk 7]
```diff
@@ -1,3 +1,5 @@
 <h1>Привет, Rails</h1>
 
-<%= link_to 'List of stations', railway_stations_path %>
\ No newline at end of file
+<%= link_to 'List of stations', railway_stations_path %>
+<%= link_to 'List of trains', trains_path %>
+<%= link_to 'List of routes', routes_path %>
```
[the remark this answers is not in the document]
Comment (Naumych) [reply app/views/welcome/index.html.erb]: Fixed

**Review by svidlonix — APPROVED** [review]

**Pull request merged**